### PR TITLE
les: fix crasher in NodeInfo when running as server

### PR DESCRIPTION
**Pull request by fjl**
The ProtocolManager computes CHT and Bloom trie roots by asking the
indexers for their current head. It tried to get the indexers from
LesOdr, but no LesOdr instance is created in server mode.

Attempt to fix this by moving the indexers, protocol creation and
NodeInfo to a new lesCommons struct which is embedded into both server
and client.

All this setup code should really be cleaned up, but this is just a
hotfix so we have to do that some other time.

### CODE REVIEW

File: les/backend.go [hunk 1]
```diff
@@ -80,6 +80,14 @@ type LightEthereum struct {
 	wg sync.WaitGroup
 }
 
+// lesCommons contains needed by both server and client.
+type lesCommons struct {
+	config                       *eth.Config
+	chainDb                      ethdb.Database // Block chain database
+	protocolManager              *ProtocolManager
+	chtIndexer, bloomTrieIndexer *core.ChainIndexer
+}
```
Comment (karalabe) [inline les/backend.go]: Could you separate out the `lesCommons` struct and its methods into a `commons.go` file (perhaps also the `NodeInfo`)? The functions seem to be scattered all over the file here and it's hard to follow.

Comment (fjl) [reply les/backend.go]: TBH they're just all at the end of the file ;)

**Comment by fjl** [comment]
PTAL

File: les/commons.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	Difficulty *big.Int                `json:"difficulty"` // Total difficulty of the host's blockchain
+	Genesis    common.Hash             `json:"genesis"`    // SHA3 hash of the host's genesis block
+	Config     *params.ChainConfig     `json:"config"`     // Chain configuration for the fork rules
+	Head       common.Hash             `json:"head"`       // SHA3 hash of the host's best owned block
+	CHT        light.TrustedCheckpoint `json:"cht"`        // Trused CHT checkpoint for fast catchup
+}
+
+// makeProtocols creates protocol descriptors for the given LES versions.
+func (c *lesCommons) makeProtocols(versions []uint) []p2p.Protocol {
+	protos := make([]p2p.Protocol, len(ClientProtocolVersions))
```
Comment (karalabe) [inline les/commons.go]: `ClientProtocolVersions` -> `versions` ?

**Comment by karalabe** [comment]
@fjl ping?
